packages/m2m-gateway/src/services/purposeService.ts

```diff
@@ -1,8 +1,45 @@
+import { m2mGatewayApi } from "pagopa-interop-api-clients";
+import { WithLogger } from "pagopa-interop-commons";
+import { toM2MPurpose } from "../api/purposeApiConverter.js";
 import { PagoPAInteropBeClients } from "../clients/clientsProvider.js";
+import { M2MGatewayAppContext } from "../utils/context.js";
 
 export type PurposeService = ReturnType<typeof purposeServiceBuilder>;
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function purposeServiceBuilder(_clients: PagoPAInteropBeClients) {
-  return {};
+export function purposeServiceBuilder({
+  purposeProcessClient,
+}: PagoPAInteropBeClients) {
```

Comment:
I know it's weird, I don't know why, but this is what's causing the tests to break 🥲 
Don't destructure here and use `clients.purposeProcessClient` below
```suggestion
export function purposeServiceBuilder(clients: PagoPAInteropBeClients) {
```